rust/arrow/src/compute/kernels/take.rs

```diff
@@ -576,186 +768,255 @@ mod tests {
-        ])
-        .data();
-        // Construct offsets
-        let value_offsets = Buffer::from(&[0, 3, 6, 7, 9].to_byte_slice());
-        // Construct a list array from the above two
-        let list_data_type = DataType::List(Box::new(DataType::Int32));
-        let list_data = ArrayData::builder(list_data_type.clone())
-            .len(4)
-            .add_buffer(value_offsets)
-            .null_count(0)
-            .null_bit_buffer(Buffer::from([0b10111101, 0b00000000]))
-            .add_child_data(value_data)
-            .build();
-        let list_array = Arc::new(ListArray::from(list_data)) as ArrayRef;
-
-        // index returns: [[null], null, [-1,-2,3], [2,null], [0,null,0]]
-        let index = UInt32Array::from(vec![Some(2), None, Some(1), Some(3), Some(0)]);
+        test_take_list_with_value_nulls!(i32, List, ListArray);
+    }
 
-        let a = take(&list_array, &index, None).unwrap();
-        let a: &ListArray = a.as_any().downcast_ref::<ListArray>().unwrap();
+    #[test]
+    fn test_take_large_list_with_value_nulls() {
+        test_take_list_with_value_nulls!(i64, LargeList, LargeListArray);
+    }
 
-        // construct a value array with expected results:
-        // [[null], null, [-1,-2,3], [5,null], [0,null,0]]
-        let expected_data = Int32Array::from(vec![
-            None,
-            Some(-1),
-            Some(-2),
-            Some(3),
-            Some(5),
-            None,
-            Some(0),
-            None,
-            Some(0),
-        ])
-        .data();
-        // construct offsets
-        let expected_offsets = Buffer::from(&[0, 1, 1, 4, 6, 9].to_byte_slice());
-        // construct list array from the two
-        let expected_list_data = ArrayData::builder(list_data_type)
-            .len(5)
-            .null_count(1)
-            // null buffer remains the same as only the indices have nulls
-            .null_bit_buffer(index.data().null_bitmap().as_ref().unwrap().bits.clone())
-            .add_buffer(expected_offsets)
-            .add_child_data(expected_data)
-            .build();
-        let expected_list_array = ListArray::from(expected_list_data);
+    #[test]
+    fn test_test_take_list_with_nulls() {
+        test_take_list_with_nulls!(i32, List, ListArray);
+    }
 
-        assert!(a.equals(&expected_list_array));
+    #[test]
+    fn test_test_take_large_list_with_nulls() {
+        test_take_list_with_nulls!(i64, LargeList, LargeListArray);
     }
 
     #[test]
-    fn test_take_list_with_list_nulls() {
-        // Construct a value array, [[0,null,0], [-1,-2,3], null, [5,null]]
-        let value_data = Int32Array::from(vec![
-            Some(0),
-            None,
-            Some(0),
-            Some(-1),
-            Some(-2),
-            Some(3),
-            Some(5),
-            None,
-        ])
-        .data();
+    #[should_panic(expected = "index out of bounds: the len is 4 but the index is 1000")]
+    fn test_take_list_out_of_bounds() {
```

Comment:
This test is new, but it is testing existing behaviour, not new behaviour. I added it here, since I almost caught myself breaking this behaviour in an earlier draft of this PR.